portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -0,0 +1,97 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/** Servlet that returns comments */
+@WebServlet("/comment")
+public class CommentServlet extends HttpServlet {
+
+  /** Comment data structure */
+  protected class Comment{
+    private final String text;
+    private final String user;
+    private final long timestamp;
+
+    public Comment(String user, String text, long timestamp){
+      this.text = text;
+      this.user = user;
+      this.timestamp = timestamp;
+    }
+
+  }
+
+  /** Responds a JSON with all the comments in DataStore in descending chronological order */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<Comment>();
+
+    for(Entity comment : results.asIterable()){
+      String user = (String) comment.getProperty("user");
+      String text = (String) comment.getProperty("text");
+      long timestamp = (long) comment.getProperty("timestamp");
+
+      comments.add(new Comment(user, text, timestamp));
+    }
+    
+    Gson gson  = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
+  }
+
+  /** Creates a new comment given the message and author */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    String comment = request.getParameter("comment");
+    String user = "JonathanHardcoded";
+    long timestamp = System.currentTimeMillis();
+
+    if(comment != null && comment != "" && user != null && user != ""){
+      Entity commentEntity = new Entity("comment");
+      commentEntity.setProperty("user", user);
+      commentEntity.setProperty("text", comment);
+      commentEntity.setProperty("timestamp", timestamp);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
+
+      //TODO: Desired functionality is not to reload, check the onsubmit on js.
```

Comment:
Nit: For this and other TODOS and comments. The google style guide says to leave a space between // and the next word, so // TODO instead of //TODO

Reply:
Done, also had that mistake on js files.